dqvm/src/main.lisp

```diff
@@ -74,22 +73,18 @@
     (show-welcome))
 
   (when execute
-    ;; Do all file handling here, not in the node code.
     (when (master-node-p)
-      (setf *program* (safely-read-quil execute)))
-
-    (assert (and (plusp +worker-count+)
-                 (power-of-two-p +worker-count+))
-            ()
-            "There has to be a positive power-of-two ~
+      (assert (power-of-two-p +worker-count+)
```

Comment:
I think this `assert` was previously not guarded by the `(when (master-node-p) ...)` no?  I assume this change was intentional, but mentioning it just in case.

Reply:
Yes, this is better IMO. Before, the error message would appear once for every MPI process instantiated.

Reply:
I assume this previously prevent the workers from starting if `+worker-count+` was invalid?  Will this now result in zombie workers, or does MPI handle the cleanup for you if the master node dies? (You can probably tell I have no idea how MPI actually works :-D)

Reply:
It seems MPI does indeed do the Right Thing (TM). From the [mpirun man page](https://www.open-mpi.org/doc/v3.0/man1/mpirun.1.php#sect16):

> Process Termination / Signal Handling
> During the run of an MPI application, if any process dies abnormally (either exiting before invoking MPI_FINALIZE, or dying as the result of a signal), mpirun will print out an error message and kill the rest of the MPI application.